### ISSUE TITLE
[Stock-Browser4.0/Android4.0] with cookies disabled, loading /signup does not result in 'requires cookies' screen

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#1542]
STR
1. Use stock browser on android 4.0
2. Disable cookies in that browser
3. Go to https://diresworb.org/signup (or /signin, etc.)

Expected: 'requires cookie' screen is show and form is not available
Actual: form is available (although trying to submit with it does produce the 'requires cookies' error screen).

Notes:
- attempting to fill out and submit the form does result in the 'requires cookies' screen. So cookies are indeed disabled.
- I have reproduced this on a Google Nexus (Browser/4.0.4, Android/4.0.4) and an Asus Transformer (Browser/4.0.3, Android/4.0.3)
- with cookies enabled, Stock 4.0 does pass the unit tests.
- On a Droid2, Stock 2.2/Android 2.2 immediately shows the 'requires cookies' screen when cookies are disabled.  
- Fennec browsers (12-15), on the same two 4.0 devices, immediately shows the 'requires cookies' screen when cookies are disabled.
- Chrome Beta (0.18.4409.2396), on the same two 4.0 devices, immediately shows the 'requires cookies' screen when cookies are disabled.

**Comment by jrgm** [comment]
Oh, forgot to note: the same thing happens for the /sign_in dialog with Stock/4.0 and cookies disabled; a usable form is shown, but an attempt to submit that form results in a 403 response from the wsapi call and the 'requires cookies' error screen is shown.

**Comment by shane-tomlinson** [comment]
@jrgm - this is a known limitation with regard to recent versions of Android. https://github.com/mozilla/browserid/blob/dev/resources/static/shared/network.js#L641-643.  @lloyd, @benadida and I discussed various approaches to fix this including making an extra request to the server to check whether cookies were set, but in the end it was decided the delayed failure mode here was acceptable.

I will update the referenced comment to include Android 4.0

Closing as "won't fix"
